app/v1/rand/genesis.go

```diff
@@ -0,0 +1,53 @@
+package rand
+
+import (
+	"fmt"
+	"strconv"
+
+	sdk "github.com/irisnet/irishub/types"
+)
+
+// InitGenesis stores genesis data
+func InitGenesis(ctx sdk.Context, k Keeper, data GenesisState) {
+	for height, requests := range data.PendingRandRequests {
+		for _, request := range requests {
+			h, err := strconv.ParseInt(height, 10, 64)
+			if err != nil {
+				continue
+			}
+
+			reqID := GenerateRequestID(request)
+			k.EnqueueRandRequest(ctx, h, reqID, request)
+		}
+	}
+}
+
+// ExportGenesis outputs genesis data
+func ExportGenesis(ctx sdk.Context, k Keeper) GenesisState {
+	pendingRequests := make(map[string][]Request)
+
+	k.IterateRandRequestQueue(ctx, func(height int64, request Request) bool {
+		leftHeight := fmt.Sprintf("%d", height-ctx.BlockHeight()+1)
+		pendingRequests[leftHeight] = append(pendingRequests[leftHeight], request)
+
```

Comment:
If there are some different address request a random number in the same block, pendingRequests only contains one address's request.